### PR TITLE
Update bem menu

### PR DESCRIPTION
**Pull request by dingo-d**
Still needs to be tested and tweaked.

Added a JS that will make it accessible (tabbable), modified the structure to conform to the BEM structure, and work with the updated BEM menu walker.

### CODE REVIEW

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:97]: "Unhover" effect should probably have a delay on it to help with accessibility, see https://www.w3.org/WAI/tutorials/menus/flyout/#fly-out-functionality

```suggestion
		this.mainMenu.addEventListener('mouseleave', setTimeout(() => this.tabCloseMenu(), 500));
```

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:28]: Shouldn't this be `this.mainMenu.querySelector(`.${this.IS_FOCUSED_CLASS}`);`? (since there can only be 1 thing that is focused at a time).

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:20]:
```suggestion
		return el.classList.contains(cls);
```

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:16]: Why not just use `classList`? I.e. `el.classList.add(cls)`?

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:24]: Why not use just use `classList`? I.e. `el.classList.remove(cls)`

File: blocks/init/src/Blocks/components/menu/menu.php
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/menu.php:35]: Shouldn't we ignore specific rules here rather than blanket ignore? I.e. `// phpcs:ignore No.Escaping.Rule`

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]
Left a few comments regarding some JS class handling, otherwise looks good!

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (dingo-d) [reply blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:28]: 🤔  That actually makes sense 👍🏼

File: blocks/init/src/Blocks/components/menu/menu.php
Comment (dingo-d) [reply blocks/init/src/Blocks/components/menu/menu.php:35]: Sure. I usually blanket ignore since it's easier in the projects 🤷🏼‍♂️

File: blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js
Comment (IvanGrginovInf) [inline blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js:28]: If you changed the above to `querySelector` now `navActive` should no longer be an array, please test it out.

Comment (dingo-d) [reply blocks/init/src/Blocks/components/menu/assets/main-menu-a11y.js]: I will test it out on the real theme in a few sec 👍🏼

**Review by IvanGrginovInf — DISMISSED** [review]

**Comment by dingo-d** [comment]
I'll merge this when I test this on an example project. So far the build process is broken so I'm blocked on this.

Ref. https://github.com/infinum/eightshift-libs/issues/144

**Comment by goranalkovic-infinum** [comment]
Closing this, as we're not using WP menus anymore, there's a new "Site navigation" and "Site footer" block to cover basic functionality.

**Comment by dingo-d** [comment]
Btw the only 'issue' with the walker in the libs was not updating the arguments (removing optional arg at the end iirc) to match the fix in the WP Core.

https://github.com/infinum/eightshift-libs/pull/373/files#diff-f7770bbdec7f1329b3dfd2de07b166174ff20669cb388f09b795e60f60ed38d7

The thing you added with site navigation and footer is messing things up for multilingual sites. 

Why didn't you start using FSE instead if you wanted to have reusable footers and headers?

**Pull request closed without merging**
